dlt/sources/helpers/transform.py

```diff
@@ -24,3 +26,31 @@ def _filter(_: TDataItem) -> bool:
         return count > max_items
 
     return _filter
+
+
+def pivot(columns: Union[str, List[str]], prefix: str) -> ItemTransformFunctionNoMeta[TDataItem]:
+    """Pivot a list of columns into a dictionary.
+
+    Args:
+        columns (Union[str, List[str]]): list of column names
+        prefix (str): prefix to add to the column names
+
+    Returns:
+        ItemTransformFunctionNoMeta[TDataItem]:
+            A function to pivot columns into a dict.
+    """
+    if isinstance(columns, str):
+        columns = [columns]
+
+    def _transformer(item: TDataItem) -> TDataItem:
+        """Pivot columns into a dictionary.
+
+        Args:
+            item (TDataItem): a data item.
+
+        Returns:
+            TDataItem: a data item with pivoted columns.
+        """
+        return {prefix + col: item[ind] for ind, col in enumerate(columns)}
```

Comment:
when `columns` are string (you allow that in the Union) it will enumerate characters. also you didn't write a test for that

Reply:
On lines 42-43 I cast `str` to `list`, so it'll work. Added a test however.